tracer/src/Datadog.Trace.Coverage.collector/AssemblyProcessor.cs

```diff
@@ -0,0 +1,688 @@
+                                    var currentInstruction = instructions.First(i => i.Offset == currentSequencePoint.Offset);
+                                    var clonedInstruction = instructions[instructions.IndexOf(currentInstruction) + instructionsOriginalLength];
+
+                                    if (!currentSequencePoint.IsHidden)
+                                    {
+                                        totalInstructions++;
+                                        localInstructions++;
+                                        clonedInstructionsWithOriginalSequencePoint.Add(Tuple.Create(clonedInstruction, currentSequencePoint));
+                                    }
+
+                                    var clonedSequencePoint = new SequencePoint(clonedInstruction, currentSequencePoint.Document);
+                                    clonedSequencePoint.StartLine = currentSequencePoint.StartLine;
+                                    clonedSequencePoint.StartColumn = currentSequencePoint.StartColumn;
+                                    clonedSequencePoint.EndLine = currentSequencePoint.EndLine;
+                                    clonedSequencePoint.EndColumn = currentSequencePoint.EndColumn;
+                                    sequencePoints.Add(clonedSequencePoint);
+
+                                    if (string.IsNullOrEmpty(methodFileName))
+                                    {
+                                        methodFileName = currentSequencePoint.Document.Url;
+                                    }
+                                }
+
+                                var clonedInstructions = instructions.Skip(instructionsOriginalLength).ToList();
+                                var clonedInstructionsLength = clonedInstructions.Count;
+
+                                // Step 6 - Modify local var to add the Coverage Scope instance.
+                                var coverageScopeVariable = new VariableDefinition(module.ImportReference(typeof(CoverageScope)));
+                                methodBody.Variables.Add(coverageScopeVariable);
+
+                                // Step 7 - Insert initial condition
+                                if (string.IsNullOrEmpty(methodFileName))
+                                {
+                                    methodFileName = "unknown";
+                                }
+
+                                var tryGetScopeMethodRef = module.ImportReference(ReportTryGetScopeMethodInfo);
+                                instructions.Insert(0, Instruction.Create(OpCodes.Ldstr, methodFileName));
+                                instructions.Insert(1, Instruction.Create(OpCodes.Ldloca, coverageScopeVariable));
+                                instructions.Insert(2, Instruction.Create(OpCodes.Call, tryGetScopeMethodRef));
+                                instructions.Insert(3, Instruction.Create(OpCodes.Brtrue, instructions[instructionsOriginalLength + 3]));
```

Comment:
Should the index be `instructionsOriginalLength + 3 + 1`? I'm trying to visualize it in my head but maybe I'm wrong

Reply:
Well a length used as an index already has intrinsic the `+1`, the `+3` is regarding about the number of instructions before.

I can say that the code is correct because we have the snapshot tests passing 😊😊